### ISSUE TITLE
A blank line at the end of output.

### ISSUE THREAD

**Issue by zhuxiaoxi** [sharkdp/bat#379]
When using the “--style plain” parameter, there will be a blank line at the end of the output.

**Comment by sharkdp** [comment]
Thank you for reporting this!

**Comment by sharkdp** [comment]
Hm.. I think this is actually caused by `less` somehow. Everything is alright if we set `--paging=never`.

**Comment by teresaejunior** [comment]
Couldn't this be related to https://github.com/sharkdp/bat/issues/434? Maybe there is an extra newline when using with `--plain`?

**Comment by sharkdp** [comment]
> Couldn't this be related to #434? Maybe there is an extra newline when using with `--plain`?

No, I don't think it is related.

I think the problem is the following.

Say we have a file "test.md" with content "foo":
```
echo foo > test.md
```

When we output this file via bat, we see the wrong behavior (notice the additional empty line):
```
▶ bat -p test.md ; echo test 
foo

test
```

To see what `bat` actually outputs, we can pipe it's output to `bat -A` which shows all non-printable characters (we have to use `--color=always` to make sure that it still outputs ANSI code):
```
▶ bat --color=always -p test.md | bat -Ap
␛[38;2;255;255;255mfoo␊
␛[0m
```

We can see that `bat` outputs the ANSI sequence `␛[38;2;255;255;255m` to set the foreground color to white. Then, it outputs `foo␊`, i.e. the full first line of the file (including the newline / line-feed). Finally, it outputs the ANSI sequence `␛[0m` to reset the foreground color.

I think it is this additional (empty) line including the `␛[0m` sequence which trips up `less`. Again, notice that everything is fine if we disable the pager:
```
▶ bat --paging=never -p test.md ; echo test
foo
test
```

**Comment by sharkdp** [comment]
We can also confirm this by using `printf`:

```
▶ printf "\x1b[38;2;255;255;255mfoo\n\x1b[0m" | less -FRX; echo test 
foo

test

/tmp                                                                               
▶ printf "\x1b[38;2;255;255;255mfoo\n\x1b[0m"; echo test            
foo
test
```

We could potentially fix this inside `bat` by moving the ANSI reset-code before the newline:
```
▶ printf "\x1b[38;2;255;255;255mfoo\x1b[0m\n" | less -FRX; echo test
foo
test
```

**Comment by sharkdp** [comment]
Fixed in [v0.10.0](https://github.com/sharkdp/bat/releases/tag/v0.10.0).
